### ISSUE TITLE
Display multiple annotations for the same time stamp 

### ISSUE THREAD

**Issue by artemiusgreat** [d3fc/d3fc#1635]
I would appreciate a recommendation or some guidance on the best way to implement something like on the screenshot below. 
https://c.mql5.com/18/84/trades__4.png 
Specifically, there are some candles that have multiple arrows on top of them, e.g. 3 blue arrows and 2 red arrows or 2 blue arrows and 1 red arrow. 
Regular time series wouldn't allow showing multiple arrows for the same time stamp on X scale. 
https://codepen.io/artemiusgreat/pen/BaKwzKb 
Annotations could be a solution, but it looks like they also allow only one element per one value on the scale. 
**In other words, how to draw multiple vertical annotations for random values on X scale?**

**Comment by ColinEberhardt** [comment]
From looking at the image you shared, these annotations look a little bit like error bars. How about using an error bar series an using the decorate pattern to add the arrows you need?

https://github.com/d3fc/d3fc/blob/master/packages/d3fc-series/README.md#errorbar

**Comment by artemiusgreat** [comment]
As far as I can see, error bars can only have low and high value. Meanwhile, each candle or bar may have a bunch of arrows or other marks. I tried to draw another picture using candles from D3. 
https://imgur.com/a/3GqjTcI 
These arrows are supposed to demonstrate moments when the stock was bought or sold. 
Let's say, on this particular screenshot, I have a 1-hour time frame, so each candle represents 1 hour. This screenshot shows 2 possible used cases. 

**Case #1** - works as expected when arrows is a time series and I always buy or sell stock on the new candle 

I decide to buy stock and the chart shows the 1st blue arrow. 
In the next hour (candle) I want to buy more stock and the chart shows one more blue arrow. 
Then, in a couple of hours (candles) I want to sell the stock and the chart shows a black arrow. 
Case closed, that worked. 

**Case #2** - doesn't work because 4 arrows should be displayed within the same candle 

It's 3:10 pm and the stock price is rapidly falling down and I decide to sell it, the first black arrow shows up. 
I see that the stock keeps falling and I sell more, the second black arrow shows up on the same candle. 
In a couple of minutes, I sell more stock and it shows the 3rd black arrow. 
At 3:55pm I decided to close my position and buy the stock back, now it shows the blue arrow. 
Case closed, but errors bars can't display 4 arrows on 1 candle because error bar can have only 2 values.

**Comment by artemiusgreat** [comment]
Managed to create a simple example that works with integer X and Y values. 
https://codepen.io/artemiusgreat/pen/RwRoKaB

Unfortunately, an example that is closer to reality doesn't work. 
Maybe, because I don't understand how to convert price value to canvas coordinates. 
It feels like `scale.invert` returns global coordinates and `context` passed to `decorate` is using some local or relative coordinates.
https://codepen.io/artemiusgreat/pen/BazKEjw 

In the example above, each candle in the series has 5 marks with prices ranging from `point.open` to `point.close`. 
**Is there a way to properly show these marks on top of the candle?** 

Cross-reference to SO 
https://stackoverflow.com/questions/64475739/place-several-symbols-on-top-of-each-candle-on-the-canvas-chart

**Comment by artemiusgreat** [comment]
Answered on SO.
